test/path_tokenizer.h

```diff
@@ -0,0 +1,20 @@
+//
+// Created by wkl04 on 12-12-2019.
+//
+
+#ifndef MQTTSUBSCRIPTION_PATH_TOKENIZER_H
+#define MQTTSUBSCRIPTION_PATH_TOKENIZER_H
+
+#include <boost/tokenizer.hpp>
+#include <string>
+#include <boost/functional/hash.hpp>
+
+using mqtt_path_tokenizer_t = boost::tokenizer< boost::char_separator<char>, MQTT_NS::string_view::const_iterator, std::string >;
+
+static inline mqtt_path_tokenizer_t mqtt_path_tokenizer(MQTT_NS::string_view const &path)
+{
+    static boost::char_separator<char> mqtt_path_separator("/",  "", boost::keep_empty_tokens);
+    return mqtt_path_tokenizer_t(path, mqtt_path_separator);
```

Comment:
I don't see any indication in the implementation of boost::char_separator<> that the type can be re-used, or accessed from multiple threads at once.

As such I think it's required that the user of this function, instead of using this function, create their own char_seperator, and path tokenizer.

Reply:
At my local code I already switched the tokenizer to a boost spirit x3 parser, which is faster. So I will commit this soon.

Reply:
But regarding multithreading, the subscription map itself is already a shared resource. So if this needs to be used from multiple threads, this needs to be locked.

Reply:
Sure, spirit parser sounds fine.

My point wasn't just multithreading though. It was that the function can never be called twice, because you'll get a char seperator in an unknown state.